### ISSUE TITLE
Coll.find_one() sorted

### ISSUE THREAD

**Issue by mikeckennedy** [BeanieODM/beanie#207]
Hey. I would love to do the following, more simply:

```python
fruits = await Product.find_all(Product.category == 'fruit').skip(0).limit(1).sort(-Product.price).to_list()
most_expensive_fruit = None
if fruits:
    most_expensive_fruit = fruits[0]
```

Maybe I'm missing how to do this with the current API. But I'd like something equivalent to above that is a one-liner:

```python
most_expensive_fruit = await Product.find(Product.category == 'fruit').sort(-Product.price).first_or_none()
```

Is there a way I haven't seen in the docs to do this?

**Comment by roman-right** [comment]
Hey,
This totally makes sense. I'll add this soon. 
I will let you know when this will be in place.

**Comment by mikeckennedy** [comment]
Super, thanks a bunch!

**Comment by mikeckennedy** [comment]
How's this one looking. I think if you jam this method on to the FindMany class (or whatever comes out of sort), then you'd basically be done:

```python
async def first_or_none(async_iterable):
    if not async_iterable:
        return None

    async for item in async_iterable:
        return item

    return None
```

**Comment by roman-right** [comment]
Thank you! This method is available in 1.9.2 now
Here is the api doc https://roman-right.github.io/beanie/api-documentation/query/#findmanyfirst_or_none

**Comment by roman-right** [comment]
And here is the example: https://roman-right.github.io/beanie/tutorial/finding-documents/#finding-documents
